src/DependencyInjection/LazyContainerFactory.php

```diff
@@ -106,10 +107,15 @@ static function (Container $container): EasyCodingStandardStyle {
         $configFiles = [__DIR__ . '/../../config/config.php', ...$configFiles];
 
         foreach ($configFiles as $configFile) {
-            $configClosure = require $configFile;
-            Assert::isCallable($configClosure);
-
-            $configClosure($ecsConfig);
+            $configBuilder = require $configFile;
+            Assert::isCallable($configBuilder);
+            Assert::isInstanceOf(
+                $configBuilder,
+                ECSConfigBuilder::class,
+                "Invalid configuration (use ECSConfig::configure()): {$configFile}"
+            );
```

Comment:
that's not necessary imo, as `ECSConfigBuilder` is syntax sugar, and forcing it may cause invalid result, eg on conditional inside config file, eg:

```php
return static function (ECSConfig $ecsConfig) {
    if (rand(0,1 )) {
        // do something...
   }
};
```

even this valid on this `$configFiles` use case, that may cause change in the future on some config.

Reply:
Also, load `$configFiles` with `if condition` inside condition will BC break for ecs extension, so it can't be applicable.

Reply:
The issue is that if you run the old style config with newer ecs, then it doesn't work properly:
- It loads your config file twice and sets settings when loading every rule set
- Since `SimpleParameterProvider` stores settings statically, it gets duplicated entries or overriden settings

Reply:
then that need to be fixed instead of forcing to use `ECSConfigBuilder`.

Reply:
You are basically forced to use it anyway since there is implicit assumption that configuration object calling is idempotent, which is not the case for `ECSConfig`, since it modifies static state. This PR just makes the assumption explicit.

Reply:
it is not, it `callable` to be invoked, and `return static function(ECSConfig)` vs `return ECSConfig::configure()->()` should be equal result. Your change make BC break and make conditional inside custom closure config no longer working.

Again, please create failing test case or repository that demonstrate your issue, that should be fixed there instead of forcing to use `ECSConfigBuilder`, thank you.

Reply:
For reference, the config object is called here:
- `src/DependencyInjection/LazyContainerFactory.php:118`
- `src/Caching/FileHashComputer.php:25`